### PR TITLE
[TASK-18] As a user, I can see NPS types of question patterns in surveys

### PR DESCRIPTION
**Pull request by rafayet-monon**
Task Card - https://github.com/rafayet-monon/surveyor/projects/1#card-45413295

## What happened
Created Net promoting score type survey question components.
 
## Insight
- [x] NPS question component and style
- [x] Implement styles 

## Proof Of Work

nps

### CODE REVIEW

File: src/stylesheets/components/_np-score.scss [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    opacity: 0.5;
+
+    transition: color 200ms;
+  }
+
+  &__number-container:hover > &__number-label &__number {
+    opacity: 1;
+  }
+
+  &__number-container > &__number-label:hover ~ &__number-label &__number {
```
Comment (rafayetn) [inline src/stylesheets/components/_np-score.scss:50]: - :warning: Expected ".np-score__number-container > .np-score__number-label:hover ~ .np-score__number-label .np-score__number" to have no more than 3 compound selectors (selector-max-compound-selectors)

File: src/screens/SurveyDetail/question.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 
-import Rating from 'components/Rating';
+import NetPromotingScore from 'components/NetPromotingScore';
```
Comment (Lahphim) [inline src/screens/SurveyDetail/question.js:3]: @rafayet-monon As I comment here (https://github.com/rafayet-monon/surveyor/pull/20/files#r544779215), I think once we improve that part then we don't have to remove the the previous codebase like this ✨

Comment (carryall) [inline src/screens/SurveyDetail/question.js:3]: @rafayet-monon  The same comment @Lahphim added on [this PR](https://github.com/rafayet-monon/surveyor/pull/20/files#r544779215) 🙌🏼

File: src/stylesheets/components/_np-score.scss [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    font-size: rem(20px);
+    line-height: rem(25px);
+
+    cursor: pointer;
+    opacity: 0.5;
+
+    transition: color 200ms;
+  }
+
+  &__number-container:hover > &__number-label &__number {
```
Comment (Lahphim) [inline src/stylesheets/components/_np-score.scss]: @rafayet-monon We can group these opacity block 🙌

File: src/stylesheets/components/_np-score.scss [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    opacity: 1;
+  }
+
+  &__number-container > &__number-label:hover ~ &__number-label &__number { /* stylelint-disable-line */
+    opacity: 0.5;
+  }
+
+  &__number--selected {
+    opacity: 1;
+  }
```
Comment (carryall) [inline src/stylesheets/components/_np-score.scss]: @rafayet-monon  How about like this ?
```suggestion
  &__number {
    margin: rem(15px 13px);

    color: map-get($brand-white, 'primary');
    font-weight: map-get($font-weight, 'bolder');
    font-size: rem(20px);
    line-height: rem(25px);

    cursor: pointer;

    transition: color 200ms;
  }
  
  &__number--selected,
  &__number-container:hover > &__number-label &__number {
    opacity: 1;
  }

  &__number,
  &__number-container > &__number-label:hover ~ &__number-label &__number { /* stylelint-disable-line */
    opacity: 0.5;
  }
```

File: src/stylesheets/components/_np-score.scss [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+  &__number-container {
+    width: fit-content;
+    height: rem(56px);
+    margin: 0 auto;
+    padding: 0;
+
+    list-style: none;
+
+    border: 0.5px solid map-get($brand-white, 'primary');
```
Comment (Lahphim) [inline src/stylesheets/components/_np-score.scss]: @rafayet-monon I think `0.5px` has the same display like `1px` 🤔

File: src/stylesheets/components/_np-score.scss [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    color: map-get($brand-white, 'primary');
+    text-align: start;
+  }
+
+  &__high-score {
+    @include make-col(6);
+
+    color: map-get($brand-white, 'primary');
+    text-align: end;
+  }
```
Comment (carryall) [inline src/stylesheets/components/_np-score.scss]: @rafayet-monon How about we group it like this?

```suggestion
  &__low-score,
  &__high-score {
    @include make-col(6);

    color: map-get($brand-white, 'primary');
  }

  &__low-score {
    text-align: start;
  }

  &__high-score {
    text-align: end;
  }
```

File: src/components/NetPromotingScore/index.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+
+const NetPromotingScore = ({ start, end }) => {
+  const startScore = start || 1;
+  const endScore = end || 10;
+  const [nps, setNps] = useState(Math.ceil((startScore + endScore) / 2));
```
Comment (carryall) [inline src/components/NetPromotingScore/index.js]: @rafayet-monon How about just let the default value be 0 🤔  what do you think?

File: src/stylesheets/components/_np-score.scss [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    display: inline-block;
+    height: rem(55px);
+    margin: 0;
+
+    border-right: 0.5px solid map-get($brand-white, 'primary');
+  }
+
+  &__number-label--no-border {
+    border-right: 0;
+  }
```
Comment (carryall) [inline src/stylesheets/components/_np-score.scss]: @rafayet-monon How about using the `:last-of-type` instead ? so we can remove the class `&__number-label--no-border`
```suggestion
  &__number-label:last-of-type {
    border-right: 0;
  }
```

File: src/components/NetPromotingScore/index.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+                    ? 'np-score__number np-score__number--selected'
+                    : 'np-score__number'
+                }
+              >
+                { score }
+              </p>
+            </li>
+          );
+        }) }
+        <div className="np-score__identifier">
```
Comment (carryall) [inline src/components/NetPromotingScore/index.js]: @rafayet-monon  Should we move this out of the `ul` so when the mouse hovered on them the score opacity won't change
npshover

**Comment by rafayet-monon** [comment]
@Lahphim @carryall Fixed reviews in [0bb6ceb](https://github.com/rafayet-monon/surveyor/pull/21/commits/0bb6ceb38a273c94d5a73c7efc4d7294ec2068a9)

File: src/stylesheets/components/_np-score.scss [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+    transition: color 200ms;
+  }
+
+  &__number-container:hover > &__number-label &__number,
+  &__number--selected {
+    opacity: 1;
+  }
+
+  /* stylelint-disable */
```
Comment (rafayetn) [inline src/stylesheets/components/_np-score.scss:48]: - :warning: Expected ".np-score__number-container > .np-score__number-label:hover ~ .np-score__number-label .np-score__number" to have no more than 3 compound selectors (selector-max-compound-selectors)

**Review by Lahphim — APPROVED** [review]

**Review by carryall — APPROVED** [review]

**Pull request merged**